extensions/security-webauthn/runtime/src/main/java/io/quarkus/security/webauthn/WebAuthnRecorder.java

```diff
@@ -34,16 +33,15 @@ public WebAuthnRecorder(RuntimeValue<HttpConfiguration> httpConfiguration, Runti
 
     public void setupRoutes(BeanContainer beanContainer, RuntimeValue<Router> routerValue, String prefix) {
         WebAuthnSecurity security = beanContainer.beanInstance(WebAuthnSecurity.class);
-        WebAuthnAuthenticationMechanism authMech = beanContainer.beanInstance(WebAuthnAuthenticationMechanism.class);
-        IdentityProviderManager identityProviderManager = beanContainer.beanInstance(IdentityProviderManager.class);
-        WebAuthnController controller = new WebAuthnController(security, config.getValue(), identityProviderManager, authMech);
+        WebAuthnController controller = new WebAuthnController(security);
         Router router = routerValue.getValue();
         BodyHandler bodyHandler = BodyHandler.create();
         // FIXME: paths configurable
         // prefix is the non-application root path, ends with a slash: defaults to /q/
+        router.post(prefix + "webauthn/login-challenge").handler(bodyHandler).handler(controller::loginChallenge);
```

Comment:
This might be a matter of preference, but I prefer 'login-options' or 'login/options' than 'login-challenge'. The returned value is of type `PublicKeyCredentialCreationOptions`, which contains not only the webauthn challenge value but also other option values.

Reply:
Yeah, this is strictly true, from a webauthn point of view, but it has a side-effect of creating and associating a challenge in the user's session. And from the point of view of users who are new to webauthn this endpoint has caused a lot of confusion in the past. I think it's useful to make it clear that there's a pretty big side-effect here (generating and storing the challenge) and that it's just just about options. It also implies the call is mandatory (which it is, contrary to just getting options), which helps clarify what it does to users.

We could name it `login-options-and-challenge` but I fear that's too long.
I could be convinced to change it to `login-options` and improve the docs to make it super obvious that this call is required for obtaining a challenge, but I'm not sure at this point.
Perhaps others have opinions?

Reply:
> Perhaps others have opinions?

I don't think my opinion should matter here as I hardly ever used WebAuth, but here it goes: I thought `challenge` is what it does, server sends challenge and user signs it? It's term I have seen in WebAuth for dummies, in which case it seems about right.

Reply:
This is a tricky issue. It is true that "options" could indeed sound optional,but the WebAuthn specification has a method, `PublicKeyCredential.parseCreationOptionsFromJSON(options)`, for converting JSON strings into `PublicKeyCredentialCreationOptions` object. For users seeking an endpoint that returns JSON for this method, "options" might make the most sense. 
Furthermore, other libraries and specifications also use "options" for this kind endpoints:
- "Server Requirements and Transport Binding Profile", though no longer active specification, named the endpoint '/attestation/options'. 
https://fidoalliance.org/specs/fido-v2.0-rd-20180702/fido-server-v2.0-rd-20180702.html#registration

- Spring Security Passkeys named the endpoint '/webauthn/register/options'.
  https://docs.spring.io/spring-security/reference/6.4/servlet/authentication/passkeys.html#passkeys-register-options

- In the demo of fido2-net-lib, a commonly used WebAuthn library in .NET, named the endpoint '/makeCredentialOptions'.
https://github.com/passwordless-lib/fido2-net-lib/blob/master/Demo/Controller.cs#L29

Considering that the returned challenge should not be generated each time, it seems preferable to name the endpoint returning PublicKeyCredentialCreationOptions as "options" rather than "challenge."

Reply:
I recalled that Chrome engineers have proposed adding an endpoint to the WebAuthn specification dedicated to obtaining only the challenge. 
https://github.com/w3c/webauthn/issues/2152

For this endpoint, it would be better to reserve the name 'challenge' and we should use 'options' here instead.

Reply:
Yes, I think it would be fine if `.../challenge` or similar are reserved for future use and `.../register-challenge` is renamed to something like `.../register-options-challenge` or another.

However, I still prefer `.../register-options`. This is because users who can notice that an endpoint with 'challenge' in the name must be called for the challenge-response mechanism, will also likely understand from the `challenge` parameter of `navigator.credentials.get` method that they need to fetch the 'challenge' from the server.
Conversely, if users don’t recall the challenge-response mechanism from the `challenge` parameter in `navigator.credentials.get`, they probably won’t realize the need to call an endpoint containing 'challenge' in its name. For this reason, I believe this is more a matter for documentation, and naming consistency with the returned type is more important here. But this is a matter of preference.

Reply:
I've renamed them to `register-options-challenge` and `login-options-challenge`